doc/iterations/2024.5/shapes/guarantee_view_models_compatibility_between_versions.adoc

```diff
@@ -0,0 +1,39 @@
+= Add the support for view models migration
+
+== Context
+
+Users want to upgrade the version of Sirius Web without losing the models they created with an older version.
+
+== Key Result
+
+When a user creates a view model with a version n of Sirius Web, the model created must be available with any version>n of Sirius Web. 
+
+The user must also be able to contribute on how to handle evolutions on his own models to guarantee compatibility between several versions.
+
+== Solution - View models
+
+1. Each model must be identified by a version number.
+
+2. If a model version is inferior to the current version of the models used by Sirius Web then the model must be transformed silently during loading to ensure compatibility. 
```

Comment:
Lastest version of a migration participant not current version of the model.

Think of it this way I could change how we interpret the view model without breaking its content but I could write a migration participant to change previous view models to migrate them to the new interpretation. The meta model would not change one bit but models would still be migrated. For example, if we changed how we interpret the lack of an expression in old models, we could migrate them with a new default value written explicitly in the expression.